docs/release-checklist.rst

```diff
@@ -83,7 +83,7 @@ Create Branch and Apply Updates
   - summarize major changes
   - commit it
 
-- update "nix/tahoe-lafs.nix"
+- update "default.nix"
```

Comment:
Is this still necessary now? (And maybe it could say _what_ to update...)

Reply:
Yea unfortunately, as far as I can tell - at least for now.  One easy way to get rid of it is if we manually manage the version number in a file somewhere instead of using setuptools_scm - but that doesn't seem like much of a win.

Possibly there is a way to improve the nix expression so it can figure this out on its own but I don't know exactly how yet and it's probably more work than we want in this branch anyway.